### PR TITLE
Tpool testing

### PR DESCRIPTION
**Pull request by DavidVorick**
Some testing.

Most notable: changes to server_test.go, Makefile

everything else is pretty standard and not-fancy. tpool test coverage is nearly at 80% already, personally I think it's good enough for the present.

I'm now going to be working on the wallet.

### CODE REVIEW

File: modules/transactionpool/transactionpool.go [hunk 1]
```diff
@@ -58,19 +63,17 @@ type (
...
 	// Check that the input modules are non-nil.
 	if cs == nil {
-		err = errors.New("transaction pool cannot use a nil state")
-		return
+		return nil, errNilCS
 	}
 	if g == nil {
-		err = errors.New("transaction pool cannot use a nil gateway")
-		return
+		return nil, errNilGateway
```
Comment (lukechampine) [inline modules/transactionpool/transactionpool.go]: these checks still feel unnecessary to me.

**Comment by lukechampine** [comment]
looks good, I think splitting up the unit/integration tests was a good idea and we came up with a clean way of implementing it.

File: modules/transactionpool/transactionpool.go (hunk 1 above)
Comment (DavidVorick) [inline modules/transactionpool/transactionpool.go]: too bad :)

I think it does pay off to be meticulous. Just have to be meticulous in the right places. I do feel like 100% code coverage is one of those places.

Comment (lukechampine) [inline modules/transactionpool/transactionpool.go]: that's what I'm saying -- this is a wrong place. It's a waste of effort. Outside of testing, the `New` functions are called in exactly one place, where it is easy to verify that their inputs are non-nil. These checks are superfluous and force us to write useless tests for the sake of coverage.
If we do somehow pass `nil` to `New`, the result will almost always be an immediate panic. The risk of such a bug making it into a release is minute, and would be discovered quickly by users anyway.
I would give some examples of where nil arguments are far more likely, but I'm afraid you would want to add checks for all of those too.

**Pull request merged**